### PR TITLE
[ESP32] fix compile error for persistent-storage app

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
Compiling for persistent-storage app fails.
```
error: format '%ld' expects argument of type 'long int', but argument 5 has type 'CHIP_ERROR' {aka 'int'} [-Werror=format=]
ChipLogError(NotSpecified, "%s: FAILED %ld [%s]", #test_result, temp_test_result, chip::ErrorStr(temp_test_result));   \
                           ^~~~~~~~~~~~~~~~~~~~~                ~~~~~~~~~~~~~~~~
```
#### Change overview
fix error above

#### Testing
Compiling succeed for persistent-storage app

### CODE REVIEW

**Comment by woody-apple** [comment]
@wqx6 It sounds like something is missing from CI if there's a failure here. Can we add something to CI to catch this?

**Comment by dhrishi** [comment]
@woody-apple  Similar thing discussed here in the context of other examples https://github.com/project-chip/connectedhomeip/pull/7904#issuecomment-869416947

Will it be okay if we add all the ESP examples in CI?

**Comment by woody-apple** [comment]
> @woody-apple Similar thing discussed here in the context of other examples [#7904 (comment)](https://github.com/project-chip/connectedhomeip/pull/7904#issuecomment-869416947)
> 
> Will it be okay if we add all the ESP examples in CI?

Yes please :)

File: .github/workflows/examples-esp32.yaml [hunk 1]
```diff
@@ -78,6 +78,38 @@ jobs:
...
+                     example_binaries/$BUILD_TYPE-build/persistent-storage.elf
+            - name: Build example Shell App
+              timeout-minutes: 5
+              run: scripts/examples/esp_example.sh shell
+            - name: Copy aside build products
+              run: |
+                  mkdir -p example_binaries/$BUILD_TYPE-build
+                  cp examples/shell/esp32/build/chip-shell.elf \
+                     example_binaries/$BUILD_TYPE-build/chip-shell.elf
+            - name: Build example Temperature Measurement APP
```
Comment (dhrishi) [inline .github/workflows/examples-esp32.yaml:105]: @wqx6  Nit: APP -> App

**Review by dhrishi — APPROVED** [review]

File: .github/workflows/examples-esp32.yaml (hunk 1 above)
Comment (wqx6) [reply .github/workflows/examples-esp32.yaml]: Changed

File: examples/persistent-storage/KeyValueStorageTest.cpp [hunk 2]
```diff
@@ -36,7 +36,8 @@ using namespace chip::DeviceLayer::PersistedStorage;
         {                                                                                                                          \
             char error_str[255];                                                                                                   \
             chip::FormatCHIPError(error_str, sizeof(error_str), temp_test_result);                                                 \
-            ChipLogError(NotSpecified, "%s: FAILED %ld [%s]", #test_result, temp_test_result, chip::ErrorStr(temp_test_result));   \
+            ChipLogError(NotSpecified, "%s: FAILED %ld [%s]", #test_result, static_cast<long int>(temp_test_result),               \
```
Comment (bzbarsky-apple) [inline examples/persistent-storage/KeyValueStorageTest.cpp:39]: I think most places we use `PRId32` for CHIP_ERROR.

Though maybe we should just add a CHIP_CONFIG_* define for the right format to use for it and start using that?  Similar to how `BLE_CONFIG_ERROR_FORMAT` works.

Comment (wqx6) [reply examples/persistent-storage/KeyValueStorageTest.cpp]: Fixed

**Comment by wqx6** [comment]
> > @woody-apple Similar thing discussed here in the context of other examples [#7904 (comment)](https://github.com/project-chip/connectedhomeip/pull/7904#issuecomment-869416947)
> > Will it be okay if we add all the ESP examples in CI?
> 
> Yes please :)

@woody-apple Done.

**Review by msandstedt — APPROVED** [review]

**Comment by woody-apple** [comment]
@mspang @jmartinez-silabs @Damian-Nordic ?

**Review by bzbarsky-apple — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**